src/SSLSocket.c

```diff
@@ -552,11 +552,12 @@ int SSLSocket_setSocketForSSL(networkHandles* net, MQTTClient_SSLOptions* opts)
 	if (net->ctx != NULL || (rc = SSLSocket_createContext(net, opts)) == 1)
 	{
 		int i;
+
 		SSL_CTX_set_info_callback(net->ctx, SSL_CTX_info_callback);
```

Comment:
In this line (and later) a NULL pointer access could occur, because net->ctx could be NULL here!
This is going to be fixed in #161